datalad_fuse/fuse_.py

```diff
@@ -150,7 +150,8 @@ def getattr(self, path, fh=None):
                 # We should just fabricate stats from the key here or not even
                 # bother???!
                 lgr.debug("File not already open")
-                fsspec_file = self._adapter.open(path)
+                with self.rwlock:
```

Comment:
can we move locking down into `_adapter` and make it repository specific since we have caches per repository?! then outside code would be able to parallelize across repositories.

Reply:
@yarikoptic No, as we also need to lock whenever using a pseudo-filehandle returned from `_adapter`.

Reply:
then we might need a helper to return the appropriate (repository specific) lock for a given path to be used.